### ISSUE TITLE
3.5.0-rc1: Current directory no longer bound in some cases

### ISSUE THREAD

**Issue by pontus** [apptainer/singularity#4691]
### Version of Singularity:

What version of Singularity are you using? Run:

```
3.5.0~rc.1-1.el7
```

### Expected behavior

What did you expect to see when you do...?

### Actual behavior

What actually happend? Why was it incorrect?

With 3.4.1:

```
[pontusf@snowy-lr1 glob]$ singularity shell ../singularityhub-centos-master.img                                                                      
Singularity./home/pontusf/singularityhub-centos-master.img> pwd
/domus/h1/pontusf/glob
Singularity./home/pontusf/singularityhub-centos-master.img> ls
private
Singularity./home/pontusf/singularityhub-centos-master.img> exit
[pontusf@snowy-lr1 glob]$
```

With 3.5.0-rc1 I see instead:

```
[pontusf@snowy-lr1 glob]$ pwd
/home/pontusf/glob
[pontusf@snowy-lr1 glob]$ singularity shell ../singularityhub-centos-master.img                                                                      
Singularity./home/pontusf/singularityhub-centos-master.img> pwd
/var/local/pontusf
Singularity./home/pontusf/singularityhub-centos-master.img> ls
rpmbuild  singularity-3.0.2.tar.gz
Singularity./home/pontusf/singularityhub-centos-master.img> exit
exit
[pontusf@snowy-lr1 glob]$
```
The user pontusf has a local home direcotry (under /var/local) but if I try with a user with home directory as /home/SOMETHING, /home/SOMETHING is mapped but not the current directory if it's  under /home/ELSEWHERE.

Home directories are provided through a NFS mount and this system uses

```
[pontusf@snowy-lr1 ~]$ ls -ld /home
lrwxrwxrwx 1 root root 9  2 okt  2018 /home -> /domus/h1
[pontusf@snowy-lr1 ~]$
```
to fix the path.

If I instead use the physical path, things seem to work as expected with 3.5.0-rc1:

```
[pontusf@snowy-lr1 pontusf]$ cd /domus/h1/pontusf
[pontusf@snowy-lr1 pontusf]$ singularity shell /home/pontusf/singularityhub-centos-master.img 
Singularity./home/pontusf/singularityhub-centos-master.img> pwd
/domus/h1/pontusf
Singularity./home/pontusf/singularityhub-centos-master.img> 
```

### Steps to reproduce this behavior

Steps needed unclear as of yet.

### What OS/distro are you running

```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

```

### How did you install Singularity

Custom built RPM.

**Comment by cclerget** [comment]
> If I instead use the physical path, things seem to work as expected with 3.5.0-rc1:
> ```
> [pontusf@snowy-lr1 pontusf]$ cd /domus/h1/pontusf
> ```

so glob` is a symbolic link right ? Also can you paste debug output too ?

**Comment by pontus** [comment]
Sorry that this was unclear,  /home is a symbolic link, glob is not.

I'll run and post debug soon.

**Comment by pontus** [comment]
[exec_debug_domus_h1.txt](https://github.com/sylabs/singularity/files/3804747/exec_debug_domus_h1.txt)
[exec_debug_home.txt](https://github.com/sylabs/singularity/files/3804748/exec_debug_home.txt)

**Comment by pontus** [comment]
Where the supplied file are stderr for the command 

```singularity --debug exec   singularityhub-centos-master.img pwd```

run from /home/pontusf as well as /domus/h1/pontusf (which is the same physical directory) where running from /home/pontusf will give the output /var/local/pontusf (which is the *actual* home directory per /etc/passwd) whereas running from /domus/h1/pontusf will yield /domus/h1/pontusf as expected.

**Comment by cclerget** [comment]
@pontus #4698 fix available, if you can give it a try ? Thanks !

**Comment by pontus** [comment]
Works great, thanks!

**Comment by dtrudg** [comment]
Closed by #4698
